src/main/java/com/courtcanva/ccfranchise/services/StaffService.java

```diff
@@ -1,25 +1,71 @@
 package com.courtcanva.ccfranchise.services;
 
+import com.amazonaws.util.Base64;
+import com.courtcanva.ccfranchise.constants.StaffStatus;
 import com.courtcanva.ccfranchise.dtos.StaffGetDto;
+import com.courtcanva.ccfranchise.exceptions.ExpiredVerificationTokenException;
+import com.courtcanva.ccfranchise.exceptions.MailingClientException;
+import com.courtcanva.ccfranchise.exceptions.NoSuchElementException;
 import com.courtcanva.ccfranchise.mappers.StaffMapper;
 import com.courtcanva.ccfranchise.models.Staff;
 import com.courtcanva.ccfranchise.repositories.StaffRepository;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+
+@Slf4j
 @Service
 @RequiredArgsConstructor
 public class StaffService {
 
     private final StaffRepository staffRepository;
     private final StaffMapper staffMapper;
+    private final MailingService mailingService;
 
     @Transactional
-    public StaffGetDto createStaff(Staff staff) {
+    public StaffGetDto createStaff(Staff staff) throws MailingClientException {
+        Staff savedStaff = staffRepository.save(staff);
+        sendVerificationEmail(savedStaff.getId());
```

Comment:
what will happen if this send email function takes too long?

Reply:
Given the discussion we had earlier this morning, I've made `MailingService.sendVerificationEmail` invoked asynchronously, this should eliminate the risks involved with running our business logics with 3rd-party services.